### ISSUE TITLE
Unit test only for Benchmark mode

### ISSUE THREAD

**Issue by jarlebring** [nep-pack/NonlinearEigenproblems.jl#78]
Is it possible to specify somehow that a testset should only be run in benchmark mode? Some of the tests on algorithms are currently tuned to have short computation time. For instance `nlar`-test is currently set to only compute one eigenvalue because it takes too long. In a benchmark-mode it would be nice to compute more eigenvalues in addition to the standard test, e.g. 
```
@onlybench @testset "manyeigs" begin 
       nlar(nep,Neigs=5)
end
```
Or is it possible to  achieve already now in some other way?

**Comment by maxbennedich** [comment]
Good idea. It's not currently supported, but I've added the below now in my experimental branch, which should take care of it. If `run_benchmark` is true (based on the environment variable), it will forward to the `@bench` macro. Otherwise, it won't do anything at all (the `@testset` won't even be part of the compiled code).
```julia
macro onlybench(ex)
    if run_benchmark
        :(@bench($(esc(ex))))
    end
end
```
